tests/v2/validation/fleet/provisioning/networkisolation.go

```diff
@@ -0,0 +1,231 @@
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/kubeconfig"
+	"github.com/rancher/shepherd/extensions/unstructured"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+
+	extensionpods "github.com/rancher/shepherd/extensions/workloads/pods"
+)
+
+const defaultNamespace = "default"
+
+// testPNI is a set of in-order instructions that gets the IP from a pod in an existing deployment/project/namespace,
+// then creates a new deployment in the default project/namespace which tries to ping the existing pod.
+// This errors if the new pod can ping the existing one, meaning that PNI is not working.
+// Not designed to work with a deployment already in the default namespace!
+func testPNI(client *rancher.Client, clusterName, deploymentNamespace, deploymentName string) error {
+	// get one of the fleet deployment's pod IPs to use in the ping command later
+	fleetPodNames, err := pods.GetPodNamesFromDeployment(client, clusterName, deploymentNamespace, deploymentName)
+	if err != nil {
+		return err
+	}
+
+	firstPodObject, err := pods.GetPodByName(client, clusterName, deploymentNamespace, fleetPodNames[0])
+	if err != nil {
+		return err
+	}
+
+	corePod := &corev1.Pod{}
+	err = steveV1.ConvertToK8sType(firstPodObject, corePod)
+	if err != nil {
+		return err
+	}
+
+	// create a new workload / pod to ping that's hardened-compliant
+	falseP := false
+	trueP := true
+	userID := int64(1000)
+	container := corev1.Container{
+		Name:            "test-pni" + namegenerator.RandStringLower(3),
+		Image:           "redis:7-alpine",
+		ImagePullPolicy: corev1.PullAlways,
+		VolumeMounts:    nil,
+		EnvFrom:         []corev1.EnvFromSource{},
+		Command:         []string{"ping", "-c", "1", corePod.Status.PodIP},
+		Args:            nil,
+		SecurityContext: &corev1.SecurityContext{
+			AllowPrivilegeEscalation: &falseP,
+			RunAsNonRoot:             &trueP,
+			RunAsUser:                &userID,
+			SeccompProfile: &corev1.SeccompProfile{
+				Type: corev1.SeccompProfileTypeRuntimeDefault,
+			},
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{
+					corev1.Capability("ALL"),
+				},
+			},
+		},
+	}
+	podTemplate := corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: make(map[string]string),
+		},
+		Spec: corev1.PodSpec{
+			Containers:       []corev1.Container{container},
+			RestartPolicy:    corev1.RestartPolicyAlways,
+			Volumes:          nil,
+			ImagePullSecrets: []corev1.LocalObjectReference{},
+			NodeSelector:     nil,
+		},
+	}
+
+	createdDeployment, err := deployments.CreateDeployment(client, clusterName, "test-d"+namegenerator.RandStringLower(3), defaultNamespace, podTemplate, 1) // jobs.CreateJob(tt.client, clusterName, "test-pni-d", defaultNamespace, podTemplate)
+	if err != nil {
+		return err
+	}
+	// Validation:
```

Comment:
looks like an incomplete comment, can you remove it?

Reply:
I'd actually like to leave this here